### ISSUE TITLE
Progress for web not working when download file

### ISSUE THREAD

**Issue by alex9153** [dart-lang/http#1305]
```
var client = http.Client();

		var request = http.Request("GET", Uri.parse("$root/$uri"));
	
		client.send(request).then((response) {
			double progress = 0.0;
			int receved = 0;
			
			response.stream.listen((value) {
			        receved += value.length;
			        progress = receved / contentLength;
                        }
```

Handler of response.stream.listen called once with all downloaded bytes

**Comment by brianquinlan** [comment]
Unfortunately, that is a limitation of the underlying `XMLHttpRequest` implementation. I believe that `package:fetch_client` does not have this issue.

**Comment by mraleph** [comment]
Fixed by #1401
